### ISSUE TITLE
`onHide` callback for the modal-ui

### ISSUE THREAD

**Issue by toteto** [near/wallet-selector#633]
As a developer I would like to provide callback parameter to be invoked when the modal is hidden/closed. This would allow my application to act accordingly when this event happens.

**Describe the solution you'd like**

The `ModalOptions` has callback `onHide`

```ts
export interface ModalOptions {
    // ... existing interface
    onHide: (hideReason: 'user-triggered' | 'wallet-navigation') => void
}
```

The `hideReason` is parameter to distinguish when the modal is closed because of user close action (click on the `X`) or because navigation to a wallet.

As alternative approach, this method can be triggered only when the close button is clicked.

**Describe alternatives you've considered**

Listening to mutations on the DOM tree for the modal container element. While this works
- the implementation can be fragile if the internal implementation changes, it will break
- gets false-triggered when the user selects wallet option 

**Acceptance criteria**

- The callback is invoked with correct set of parameters when the modal for wallet picking is closed

**Comment by kujtimprenkuSQA** [comment]
Hey, @toteto thank you for raising this issue.

We created a Draft PR for this functionality, could you have a look at it and see if this is the solution you're looking for: https://github.com/near/wallet-selector/pull/642.

**Comment by AmmarHumackicSQA** [comment]
Hey @toteto can you share the status on this, and if what @kujtimprenkuSQA mentioned helped you? Thanks!

**Comment by AmmarHumackicSQA** [comment]
@toteto I will close this issue for now. Feel free to reopen it if needed. Thanks!

**Comment by toteto** [comment]
Thanks a lot! The draft interface looks good, would be nice to get it merged and released soon.

**Comment by AmmarHumackicSQA** [comment]
Hi @toteto we planned for it to be included in the next release (v7.6.0) set for today. Thanks!

**Comment by toteto** [comment]
Just tested it out, perfect 👌. Great work everyone!
